File: CodingChallengeSolutionsC++/2094. Finding 3-Digit Even Numbers/2094.Finding3-DigitEvenNumbers.cpp
#include <vector>
class Finding3DigitEvenNumbers{
public:
    std::vector<int> findEvenNumbers(std::vector<int>& digits) {
        std::vector<int> frequency(10, 0);
        for (int x = 0; x < digits.size(); ++x)
        {
            frequency[digits[x]]++;
        }

        std::vector<int> result;
        for (int j = 1; j <= 9; ++j)
        {
            if (frequency[j] == 0)
            {
                continue;
            }
            frequency[j]--;
            for (int k = 0; k <= 9; ++k)
            {
                if (frequency[k] == 0)
                {
                    continue;
                }
                frequency[k]--;
                for (int l = 0; l <= 8; l += 2)
                {
                    if (frequency[l] == 0)
                    {
                        continue;
                    }
                    result.push_back(j * 100 + k * 10 + l);
                }
                frequency[k]++;
            }
            frequency[j]++;
        }
        return result;
    }
};

File: CodingChallengeSolutionsC++/2094. Finding 3-Digit Even Numbers/2094.Finding3-DigitEvenNumbers.js
var findEvenNumbers = function (digits) {
    let freq = new Array(10).fill(0);
    for (let x = 0; x < digits.length; ++x) {
        freq[digits[x]]++;
    }

    let res = [];
    for (let j = 1; j <= 9; ++j) {
        if (freq[j] == 0) {
            continue;
        }
        freq[j]--;
        for (let k = 0; k <= 9; ++k) {
            if (freq[k] == 0) {
                continue;
            }
            freq[k]--;
            for (let l = 0; l <= 8; l += 2) {
                if (freq[l] == 0) {
                    continue;
                }
                res.push(j * 100 + k * 10 + l);
            }
            freq[k]++;
        }
        freq[j]++;
    }
    return res;
};